Dockerfile.commonwealth_base

```diff
@@ -0,0 +1,47 @@
+FROM node:20-slim AS base
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
+RUN apt-get update \
+ && apt-get install -y gnupg apt-transport-https gpg-agent curl ca-certificates
+
+ENV DATADOG_APT_KEYRING="/usr/share/keyrings/datadog-archive-keyring.gpg"
+ENV DATADOG_APT_KEYS_URL="https://keys.datadoghq.com"
+RUN sh -c "echo 'deb [signed-by=${DATADOG_APT_KEYRING}] https://apt.datadoghq.com/ stable 7' > /etc/apt/sources.list.d/datadog.list"
+RUN touch ${DATADOG_APT_KEYRING}
+RUN curl -o /tmp/DATADOG_APT_KEY_CURRENT.public "${DATADOG_APT_KEYS_URL}/DATADOG_APT_KEY_CURRENT.public" && \
+    gpg --ignore-time-conflict --no-default-keyring --keyring ${DATADOG_APT_KEYRING} --import /tmp/DATADOG_APT_KEY_CURRENT.public
+RUN curl -o /tmp/DATADOG_APT_KEY_06462314.public "${DATADOG_APT_KEYS_URL}/DATADOG_APT_KEY_06462314.public" && \
+    gpg --ignore-time-conflict --no-default-keyring --keyring ${DATADOG_APT_KEYRING} --import /tmp/DATADOG_APT_KEY_06462314.public
+RUN curl -o /tmp/DATADOG_APT_KEY_C0962C7D.public "${DATADOG_APT_KEYS_URL}/DATADOG_APT_KEY_C0962C7D.public" && \
+    gpg --ignore-time-conflict --no-default-keyring --keyring ${DATADOG_APT_KEYRING} --import /tmp/DATADOG_APT_KEY_C0962C7D.public
+RUN curl -o /tmp/DATADOG_APT_KEY_F14F620E.public "${DATADOG_APT_KEYS_URL}/DATADOG_APT_KEY_F14F620E.public" && \
+    gpg --ignore-time-conflict --no-default-keyring --keyring ${DATADOG_APT_KEYRING} --import /tmp/DATADOG_APT_KEY_F14F620E.public
+RUN curl -o /tmp/DATADOG_APT_KEY_382E94DE.public "${DATADOG_APT_KEYS_URL}/DATADOG_APT_KEY_382E94DE.public" && \
+    gpg --ignore-time-conflict --no-default-keyring --keyring ${DATADOG_APT_KEYRING} --import /tmp/DATADOG_APT_KEY_382E94DE.public
+
+RUN apt-get update && apt-get -y --force-yes install --reinstall datadog-agent
+
+COPY datadog-config/ /etc/datadog-agent/
+
+EXPOSE 8125/udp 8126/tcp
+
+FROM base AS build
+## needed for node-gyp to build
+RUN apt-get update && \
+    apt-get install -y python3 make gcc g++
+COPY . /usr/src/app
+WORKDIR /usr/src/app
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
+RUN pnpm run build
+RUN pnpm deploy --filter=commonwealth --prod /prod/commonwealth
+RUN mv /usr/src/app/packages/commonwealth/build /prod/commonwealth/build
+
+FROM base AS commonwealth
+ENV NODE_ENV=production
+COPY --from=build /prod/commonwealth /prod/commonwealth
+WORKDIR /prod/commonwealth
+RUN apt-get update && apt-get install -y curl # Needed for heroku
+ENV PORT=$PORT
+COPY ./scripts/datadog-entrypoint.sh /prod/commonwealth
+RUN chmod +x /prod/commonwealth/datadog-entrypoint.sh
```

Comment:
Did we decide that the max old space script is still needed? If so, perhaps it could be chained here at the `RUN` declaration? e.g. `./get-max-old-space-size.sh && ./datadog-entrypoint.sh`

Reply:
I'm not talking about container build time, but rather container runtime, during the `RUN` step.

Reply:
Oops, for some reason I thought that was the entrypoint 😅

What I meant was the `CMD` step in the other dockerfiles. Nonetheless, LGTM– tweaks can be made later.